miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -12,64 +12,287 @@
+    Sample sample = sampleService.get(id);
+    if (sample == null) {
+      throw new RestException("No sample found with ID: " + id, Status.NOT_FOUND);
+    } else {
+      SampleDto dto = Dtos.asDto(sample);
+      dto.writeUrls(uriBuilder);
+      return dto;
+    }
+  }
+
+  @RequestMapping(method = RequestMethod.GET, produces = { "application/json" })
+  @ResponseBody
+  public List<SampleDto> getSamples(UriComponentsBuilder uriBuilder) throws IOException {
+    List<Sample> samples = sampleService.list();
+    // return all samples
+    List<SampleDto> sampleDtos = Dtos.asSampleDtos(samples, true);
+    for (SampleDto sampleDto : sampleDtos) {
+      sampleDto.writeUrls(uriBuilder);
+    }
+    return sampleDtos;
+  }
+
+  @RequestMapping(value = "/dt", method = RequestMethod.GET, produces = { "application/json" })
+  @ResponseBody
+  public DataTablesResponseDto<SampleDto> getDTSamples(HttpServletRequest request, HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder);
+  }
+
+  @RequestMapping(value = "/dt/project/{id}", method = RequestMethod.GET, produces = { "application/json" })
+  @ResponseBody
+  public DataTablesResponseDto<SampleDto> getDTSamplesByProject(@PathVariable("id") Long id, HttpServletRequest request,
+      HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder, PaginationFilter.project(id));
+  }
+
+  @RequestMapping(value = "/dt/project/{id}/arrayed", method = RequestMethod.GET, produces = { "application/json" })
+  @ResponseBody
+  public DataTablesResponseDto<SampleDto> getDTArrayedSamplesByProject(@PathVariable("id") Long id, HttpServletRequest request,
+      HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder, PaginationFilter.project(id), PaginationFilter.arrayed(true));
+  }
+
+  @RequestMapping(method = RequestMethod.POST, headers = { "Content-type=application/json" })
+  @ResponseStatus(HttpStatus.CREATED)
+  @ResponseBody
+  public SampleDto createSample(@RequestBody SampleDto sampleDto, UriComponentsBuilder b, HttpServletResponse response) throws IOException {
+    if (sampleDto == null) {
+      log.error("Received null sampleDto from front end; cannot convert to Sample. Something likely went wrong in the JS DTO conversion.");
+      throw new RestException("Cannot convert null to Sample", Status.BAD_REQUEST);
+    }
+    Long id = null;
+    try {
+      Sample sample = buildHierarchy(sampleDto);
+      id = sampleService.create(sample);
+    } catch (ConstraintViolationException | IllegalArgumentException e) {
+      log.error("Error while creating sample. ", e);
+      RestException restException = new RestException(e.getMessage(), Status.BAD_REQUEST);
+      if (e instanceof ConstraintViolationException) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)